src/components/screens/editor/editor-screen.tsx

```diff
@@ -74,99 +70,55 @@ const EditorScreen: React.FC = () => {
-                </AdminFormTypographyField>
-              </Grid>
-              <Grid item md={ 2 }>
-                <AdminFormTypographyField>
-                  <DateRangeRounded style={{ fill: "darkgrey", marginRight: "0.5rem" }}/>
-                  { moment(version.createdAt).format("LLL") }
-                </AdminFormTypographyField>
-              </Grid>
-              <Grid item md={ 2 }>
-                <AdminFormTypographyField>
-                  <DateRangeRounded style={{ fill: "darkgrey", marginRight: "0.5rem" }}/>
-                  { moment(version.modifiedAt).format("LLL") }
-                </AdminFormTypographyField>
-              </Grid>
-              <Grid item md={ 2 }>
-                <AdminFormTypographyField>
-                  <PersonRounded style={{ fill: "darkgrey", marginRight: "0.5rem" }}/>
-                  placeholder_email
-                </AdminFormTypographyField>
-              </Grid>
-            </Grid>
-          </AdminFormListStack>
-        </ListItem>
-      );
-    });
+    setSelectedId(undefined);
+    setLoading(false);
   };
 
   /**
-   * Renders header for MetaformVersions listing
+   * If selected version is in production, navigates to FormEditorScreen, else to DraftEditorScreen
    */
-  const renderVersionListHeader = () => {
-    return (
-      <VersionListHeader container>
-        <Grid item md={ 6 }>
-          <AdminFormTypographyField sx={{ fontWeight: "bold" }}>
-            <ListRounded style={{ fill: "darkgrey", marginRight: "0.5rem" }}/>
-            { strings.editorScreen.formVersion }
-          </AdminFormTypographyField>
-        </Grid>
-        <Grid item md={ 2 }>
-          <AdminFormTypographyField sx={{ fontWeight: "bold" }}>
-            <DateRangeRounded style={{ fill: "darkgrey", marginRight: "0.5rem" }}/>
-            { strings.editorScreen.formCreatedAt }
-          </AdminFormTypographyField>
-        </Grid>
-        <Grid item md={ 2 }>
-          <AdminFormTypographyField sx={{ fontWeight: "bold" }}>
-            <DateRangeRounded style={{ fill: "darkgrey", marginRight: "0.5rem" }}/>
-            { strings.editorScreen.formModifiedAt }
-          </AdminFormTypographyField>
-        </Grid>
-        <Grid item md={ 2 }>
-          <AdminFormTypographyField sx={{ fontWeight: "bold" }}>
-            <PersonRounded style={{ fill: "darkgrey", marginRight: "0.5rem" }}/>
-            { strings.editorScreen.formLastModifier }
-          </AdminFormTypographyField>
-        </Grid>
-      </VersionListHeader>
-    );
+  const editMetaformOrDraft = async () => {
+    const id = selectedId;
+    let slug;
+    const editMetaform = metaforms.find(metaform => metaform.id === id);
+
+    if (editMetaform) {
+      slug = editMetaform.slug;
+      navigate(`${currentPath}/${slug}`);
+    } else {
+      navigate(`${currentPath}/${slug}/${id}`);
+    }
```

Comment:
Maybe use this function to return correct link to edit button instead of doing the navigation here?

Reply:
Good idea. Done.